### ISSUE TITLE
Refactor DistSQL keyword `GENERATED_KEY` to `KEY_GENERATE_STRATEGY`

### ISSUE THREAD

**Issue by RaigorJiang** [apache/shardingsphere#14830]
Hi community,

Currently, the way to configure the `key generate strategy` in DistSQL is different from the Java API and YAML:
- In the [Java API](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-jdbc/java-api/rules/sharding/#sharding-table-configuration), use `setKeyGenerateStrategy()`;
- In [YAML](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-jdbc/yaml-config/rules/sharding/), use `keyGenerateStrategy`;
- In [DistSQL](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-proxy/distsql/syntax/rdl/rule-definition/sharding/), `GENERATED_KEY` is used;

The keyword `GENERATED_KEY` affects the uniformity of the configuration.
\
So I recommend adjusting the keyword in the DistSQL syntax, to `KEY_GENERATE_STRATEGY`, just like `TABLE_STRATEGY`.

#### Before
```sql
CREATE SHARDING TABLE RULE t_order (
RESOURCES(resource_0,resource_1),
SHARDING_COLUMN=order_id,TYPE(NAME=hash_mod,PROPERTIES("sharding-count"=4)),
GENERATED_KEY(COLUMN=another_id,TYPE(NAME=snowflake,PROPERTIES("worker-id"=123)))
);
```

#### After
```sql
CREATE SHARDING TABLE RULE t_order (
RESOURCES(resource_0,resource_1),
SHARDING_COLUMN=order_id,TYPE(NAME=hash_mod,PROPERTIES("sharding-count"=4)),
KEY_GENERATE_STRATEGY(COLUMN=another_id,TYPE(NAME=snowflake,PROPERTIES("worker-id"=123)))
);
```

**Comment by yx9o** [comment]
I want to try.

**Comment by RaigorJiang** [comment]
@yx9o Thank you!

**Comment by yx9o** [comment]
Fixed #14962.
